lib/SILOptimizer/Differentiation/JVPCloner.cpp

```diff
@@ -1061,7 +1130,9 @@ class JVPCloner::Implementation final
     } else {
       tanBuf =
```

Comment:
This code (not your changes) feels very wrong in the first place... How is it a buffer? `tuple_extract` would never produce a buffer...

Reply:
I feel this could be my code.  This should be calling `setTangentValue` on the extracted value instead, since the result has an object type, not an address.  Maybe it's not tested.

Reply:
Thanks for fixing this for me!  Could you add a test case?  Maybe something like the following will trigger this code:
```swift
differential(at: 0.0) { x in
    let tuple = (x, 0.0)
    return tuple.x
}
```